File: backend/servico-reserva-fila/src/clientes/tipos.ts
// ─── Tipos das respostas dos microsserviços externos ─────────────────────────
// Representam apenas os campos necessários para validação de integridade.
// Os serviços externos podem retornar mais campos; apenas estes são mapeados.

/**
 * Dados mínimos do usuário retornados pelo Serviço de Autenticação e Usuário.
 * Rota: GET /api/usuarios/{id}
 */
export interface RespostaUsuario {
  id: string;
  nome: string;
  email: string;
}

/**
 * Dados mínimos do livro retornados pelo Serviço de Catálogo e Biblioteca.
 * Rota: GET /api/livros/{id}
 */
export interface RespostaLivro {
  id: string;
  titulo: string;
  autor: string;
  disponivel: boolean;
}

File: backend/servico-emprestimo/internal/cliente/tipos.go
package cliente

import "time"

// RespostaUsuario representa o payload público retornado pelo Serviço de Autenticação e Usuário
type RespostaUsuario struct {
	ID        string    `json:"id"`
	Nome      string    `json:"nome"`
	WhatsApp  string    `json:"whatsapp"`
	Email     string    `json:"email"`
	Permissao string    `json:"permissao"`
	Ativo     bool      `json:"ativo"`
	CriadoEm  time.Time `json:"criado_em"`
}

// RespostaLivro representa o payload de saída detalhado retornado pelo Serviço de Catálogo e Biblioteca
type RespostaLivro struct {
	ID                   interface{} `json:"id"` // Aceita número ou string para máxima flexibilidade
	Titulo               string      `json:"titulo"`
	Autor                string      `json:"autor"`
	Isbn                 string      `json:"isbn"`
	Categoria            string      `json:"categoria"`
	QuantidadeTotal      int         `json:"quantidade_total"`
	QuantidadeDisponivel int         `json:"quantidade_disponivel"`
	IDBiblioteca         interface{} `json:"id_biblioteca"`
	Ativo                bool        `json:"ativo"`
}
